### PR TITLE
explicit int -> bool conversion demo

### PR DESCRIPTION
**Pull request by shenlebantongying**

### CODE REVIEW

File: src/audiooutput.cc [hunk 1]
```diff
@@ -73,13 +73,13 @@ class AudioOutputPrivate: public QIODevice
 
   qint64 readData( char * data, qint64 len ) override
   {
-    if ( !len ) {
```
Comment (shenlebantongying) [inline src/audiooutput.cc:76]: I like this one.

`!len` is cringe.

File: src/dict/aard.cc [hunk 2]
```diff
@@ -811,10 +811,10 @@ vector< sptr< Dictionary::Class > > makeDictionaries( vector< string > const & f
         AAR_header dictHeader;
 
         df.read( &dictHeader, sizeof( dictHeader ) );
-        bool has64bitIndex = !strncmp( dictHeader.indexItemFormat, ">LQ", 4 );
-        if ( strncmp( dictHeader.signature, "aard", 4 )
-             || ( !has64bitIndex && strncmp( dictHeader.indexItemFormat, ">LL", 4 ) )
-             || strncmp( dictHeader.keyLengthFormat, ">H", 2 ) || strncmp( dictHeader.articleLengthFormat, ">L", 2 ) ) {
+        bool has64bitIndex = strncmp( dictHeader.indexItemFormat, ">LQ", 4 ) == 0;
```
Comment (shenlebantongying) [inline src/dict/aard.cc:814]: `strncmp()==0` is C trick to represent results with 3 possible states.

:-1:

File: src/dict/btreeidx.cc [hunk 3]
```diff
@@ -765,7 +765,7 @@ void BtreeIndex::antialias( wstring const & str, vector< WordArticleLink > & cha
     caseFolded = Folding::trimWhitespaceOrPunct( caseFolded );
   }
 
-  for ( unsigned x = chain.size(); x--; ) {
```
Comment (shenlebantongying) [inline src/dict/btreeidx.cc:768]: Nifty trick to combine 2 parts of `for` loop.

:-1:

File: src/audiooutput.cc [hunk 4]
```diff
@@ -73,13 +73,13 @@ class AudioOutputPrivate: public QIODevice
...
+    if ( len == 0 ) {
       return 0;
     }
 
     QMutexLocker locker( &mutex );
     qint64 bytesWritten = 0;
-    while ( len && !quit ) {
+    while ( (len != 0) && !quit ) {
```
Comment (xiaoyifang) [inline src/audiooutput.cc:82]: can `len=-1`

**Pull request closed without merging**